src/systemd.c

```diff
@@ -84,6 +107,55 @@ static int job_removed_callback(sd_bus_message *message, void *user_data, sd_bus
 	return 0;
 }
 
+static int validate_scope_slice_name(const char * const scope_name, const char * const slice_name)
+{
+	char *_scope_name = NULL;
+	int i, len, ret = 0;
+
+	len = strlen(scope_name) - 5;	/* strlen(".scope") */
```

Comment:
I think I'm guilty of this bug.  Do we need to check for underflow here?  What if the scope is "foo" or the slice is "bar"?

Reply:
@drakenclimber Thanks for the review. Sorry, I should have looked, I will spin v2 with the check for length.

Reply:
No need to apologize since I'm the one who originally wrote it :/.  Thanks for the help!